src/charm.py

```diff
@@ -210,21 +283,108 @@ def _configure(self, event: EventBase) -> None:
         if self._invalid_configs():
             return
         if not self._root_certificate_is_stored or not self._root_certificate_matches_config():
-            self._generate_root_certificate()
             self.tls_certificates.revoke_all_certificates()
+            self._clean_up_juju_secret(EXPIRING_CA_CERTIFICATES_SECRET_LABEL)
+            self._generate_root_certificate()
             logger.info("Revoked all previously issued certificates.")
             return
+        if not self._is_ca_cert_active():
+            logger.info("Renewing CA certificate")
+            self._renew_root_certificate()
+            return
```

Comment:
Does this mean we renew the ca certificate on every `_configure` hook?

Reply:
Only if the CA is not active, meaning it's secret has expired.